packages/lit-dev-content/site/tutorials/content/svg-templates/02.md

```diff
@@ -0,0 +1,295 @@
+is an arrangement of two or more elements. This demo's motif will be a
+series of text rotated around a point. The `<defs>` and `<use>`
+elements will compose a motif from a single `<text>` element.
+
+The `<defs>` element contains svg elements without rendering them. The
+example below defines a `<text>` element inside a `<defs>` element.
+Elements in `<defs>` can be referenced by an `id`.
+
+```ts
+const helloDefs = svg`
+  <defs>
+    <text id="chars">Hello defs!</text>
+  </defs>
+`;
+```
+
+To reuse the `<text>` element previously defined in `<defs>`, set the
+`href` property on a `<use>` element to `#chars` like the example below.
+
+```ts
+const helloDefs = svg`
+  <defs>
+    <text id="chars">Hello defs!</text>
+  </defs>
+  <use href="#chars"></use>
+`;
+```
+
+A key feature that makes the combination of `<defs>` and `<use>` so
+powerful is the fact that attributes and properties applied to `<use>`
+do not affect the referenced element in `<defs>`. This allows
+developers to compound properties like `transform`.
+
+In the example below, the `<text>` element in `<defs>` has no
+rotation. However, we can apply rotation to `<use>` without
+affecting the layout of the `<text>` element in `<defs>`.
+
+```ts
+const helloDefs = svg`
+  <defs>
+    <text id="chars">Hello defs!</text>
+  </defs>
+  <use
+    href="#chars"
+    transform="rotate(180, 0,0)">
+  </use>
+`;
+```
+
+Lastly, `<g>` is a special element that applies its properties to all
+child elements. This is helpful when we need to distribute a property
+to a number of elements.
+
+In the example below, the `<g>` element will apply its `transform`
+to every child element. In this case, the `<use>` element will be both
+`rotate` _and_ `translate`.
+
+```ts
+const helloGroups = svg`
+  <defs>
+    <text id="chars">Hello defs!</text>
+  </defs>
+  <g transform="translate(50, 50)">
+    <use
+      href="#chars"
+      transform="rotate(${currRotation}, 0,0)">
+    </use>
+  </g>
+`;
+```
+
+### Apply
+
+Add two new reactive properties to `repeat-pattern`: `num-prints`
+and `rotation-offset`.
+
+The `num-prints` attribute will affect how many prints are in a
+motif. The `rotation-offset` attribute will provide an
+initial rotation to the motif.
+
+{% switchable-sample %}
+
+```ts
+export class RepeatPattern extends LitElement {
+  @property({type: Number, attribute: "num-prints"}) numPrints = 7;
+  @property({
+    type: Number,
+    attribute: "rotation-offset",
+  }) rotationOffset = 0;
+  ...
+}
+```
+
+```js
+export class RepeatPattern extends LitElement {
+  static properties = {
+    chars: {type: String},
+    numPrints: {type: Number, attribute: 'num-prints'},
+    rotationOffset: {
+      type: Number,
+      attribute: 'rotation-offset',
+    },
+  };
+
+  constructor() {
+    super();
+    this.chars = 'lit';
+    this.numPrints = 7;
+    this.rotationOffset = 0;
+  }
+  ...
+}
+```
+
+{% endswitchable-sample %}
+
+Provide an `id` to the `text` element in `createElement`.
```

Comment:
Having to place these `id`'s here is kind of annoying. Maybe they should be included when the functions are defined and then just called out here?

Reply:
I feel like having the user set an id once is okay here, this one it's to help them correlate what they wrote in the "hello text" example in 01.

Adding an `id` so they understand the relationship to `<use>` is the point i feel